packages/medusa/src/api/routes/admin/stock-locations/update-stock-location.ts

```diff
@@ -0,0 +1,183 @@
+ * responses:
+ *   200:
+ *     description: OK
+ *     content:
+ *       application/json:
+ *         schema:
+ *           type: object
+ *           properties:
+ *             stock_location:
+ *               $ref: "#/components/schemas/StockLocationDTO"
+ *   "400":
+ *     $ref: "#/components/responses/400_error"
+ *   "401":
+ *     $ref: "#/components/responses/unauthorized"
+ *   "404":
+ *     $ref: "#/components/responses/not_found_error"
+ *   "409":
+ *     $ref: "#/components/responses/invalid_state_error"
+ *   "422":
+ *     $ref: "#/components/responses/invalid_request_error"
+ *   "500":
+ *     $ref: "#/components/responses/500_error"
+ */
+export default async (req: Request, res: Response) => {
+  const { id } = req.params
+
+  const locationService: IStockLocationService = req.scope.resolve(
+    "stockLocationService"
+  )
+
+  const stockLocation = await locationService.update(
+    id,
+    req.validatedBody as AdminPostStockLocationsLocationReq
+  )
+
+  res.status(200).json({ stock_location: stockLocation })
+}
+
+class StockLocationAddress {
+  @IsString()
+  address_1: string
+
+  @IsOptional()
+  @IsString()
+  address_2?: string
+
+  @IsOptional()
+  @IsString()
+  city?: string
+
+  @IsString()
+  country_code: string
+
+  @IsOptional()
+  @IsString()
+  phone?: string
+
+  @IsOptional()
+  @IsString()
+  postal_code?: string
+
+  @IsOptional()
+  @IsString()
+  province?: string
+}
+
+/**
+ * @schema AdminPostStockLocationsLocationReq
+ * type: object
+ * properties:
+ *   name:
+ *     description: the name of the stock location
+ *     type: string
+ *   address_id:
+ *     description: the stock location address ID
+ *     type: string
+ *   metadata:
+ *     type: object
+ *     description: An optional key-value map with additional details
+ *     example: {car: "white"}
+ *   address:
+ *     type: object
+ *     properties:
+ *       address_1:
+ *         type: string
+ *         description: Stock location address
+ *         example: 35, Jhon Doe Ave
+ *       address_2:
+ *         type: string
+ *         description: Stock location address' complement
+ *         example: apartment 4432
+ *       city:
+ *         type: string
+ *         description: Stock location address' city
+ *         example: Mexico city
+ *       country_code:
+ *         type: string
+ *         description: Stock location address' country
+ *         example: MX
+ *       phone:
+ *         type: string
+ *         description: Stock location address' phone number
+ *         example: +1 555 61646
+ *       postal_code:
+ *         type: string
+ *         description: Stock location address' postal code
+ *         example: HD3-1G8
+ *       province:
+ *         type: string
+ *         description: Stock location address' province
+ *         example: Sinaloa
+ *       metadata:
+ *         type: object
+ *         description: An optional key-value map with additional details
+ *         example: {car: "white"}
```

Comment:
suggestion(non-blocking): extract as separate OAS object

Reply:
referencing StockLocationAddressInput would be ok?